staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -60,6 +60,16 @@ type JobList struct {
 // JobSpec describes how the job execution will look like.
 type JobSpec struct {
 
+	// Suspend specifies whether the Job controller should create Pods or not. If
+	// a Job is created with suspend set to true, no Pods are created by the Job
+	// controller. If a Job is suspended after creation (i.e. the flag goes from
+	// false to true), the Job controller will delete all active Pods associated
+	// with this Job. Users must design their workload to gracefully handle this.
+	// This is an alpha field and requires enabling the SuspendJob feature gate.
+	// Defaults to false.
+	// +optional
+	Suspend *bool `json:"suspend,omitempty" protobuf:"varint,10,opt,name=suspend"`
```

Comment:
Question: Why do we need a pointer instead of a plain boolean?

The answer might be: because we want webhooks to be able to differentiate whether a user set a preference. In that case, we shouldn't be filling up a default in `defaults.go`, but the strategy. Otherwise webhooks get the value already filled up.

But realistically, I think webhooks will only want to enforce suspend=true, regardless of user preference, so I don't see the need for a pointer.

Reply:
don't we also need this to drop the field when the feature is disabled?

Reply:
then this again is the same question: what does it mean to disable the feature? https://github.com/kubernetes/kubernetes/pull/98727#discussion_r570467511

Reply:
it doesn't only mean disallowing the proposed behavior, but also disable the logic completely so that operators can shut it down if there is a bug or unexpected behavior in the initial implementation.

Reply:
but in that case, the suspended=false doesn't have any alterations

Reply:
Resolved, see the other thread.